assets/src/edit-story/components/library/panes/media/deleteDialog.js

```diff
@@ -0,0 +1,122 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { useCallback } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../../../app/api';
+import { Primary, Secondary } from '../../../../components/button';
+import Dialog from '../../../../components/dialog';
+import { useSnackbar } from '../../../../app/snackbar';
+import { useMedia } from '../../../../app/media';
+
+const DialogBody = styled.div`
+  width: 512px;
+  font-family: ${({ theme }) => theme.fonts.body1.family};
+  line-height: ${({ theme }) => theme.fonts.body1.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.body1.weight};
+  font-size: ${({ theme }) => theme.fonts.body1.size};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const Space = styled.div`
+  width: 8px;
+`;
+
+/**
+ * Display a confirmation dialog for when a user wants to delete a media element.
+ *
+ * @param {Object} props Component props.
+ * @param {number} props.mediaId Selected media element's ID.
+ * @param {string} props.type Selected media element's type.
+ * @param {function()} props.onClose Callback to toggle dialog display on close.
+ * @return {null|*} The dialog element.
+ */
+function DeleteDialog({ mediaId, type, onClose }) {
+  const {
+    actions: { deleteMedia },
+  } = useAPI();
+  const { showSnackbar } = useSnackbar();
+  const {
+    actions: { deleteMediaElement },
+  } = useMedia();
+
+  const onDelete = useCallback(async () => {
+    onClose();
+    try {
+      await deleteMedia(mediaId);
+      deleteMediaElement({ id: mediaId });
+    } catch (err) {
+      showSnackbar({
+        message: __('Failed to delete media item.', 'web-stories'),
+      });
+    }
+  }, [deleteMedia, deleteMediaElement, mediaId, onClose, showSnackbar]);
+
+  const imageDialogTitle = __('Delete Image?', 'web-stories');
+  const videoDialogTitle = __('Delete Video?', 'web-stories');
+  const imageDialogDescription = __(
+    'You are about to permanently delete this image from your site. ' +
+      'The image will appear broken in any WordPress content that uses it. ',
+    'web-stories'
+  );
+  const videoDialogDescription = __(
+    'You are about to permanently delete this video from your site. ' +
+      'The video will appear broken in any WordPress content that uses it. ',
+    'web-stories'
+  );
+
+  // Keep icon and menu displayed if menu is open (even if user's mouse leaves the area).
+  return (
+    <Dialog
+      open={true}
+      onClose={onClose}
+      title={type == 'image' ? imageDialogTitle : videoDialogTitle}
+      actions={
+        <>
+          <Secondary onClick={onClose}>{__('Cancel', 'web-stories')}</Secondary>
+          <Space />
+          <Primary onClick={onDelete}>{__('Delete', 'web-stories')}</Primary>
```

Comment:
Let's be consistent with the dialog actions already used in `assets/src/edit-story/components/header/buttons.js` and in #1733.

1. Let's remove the spacer
1. Let's make  the cancel button a `<Plain>` one

Reply:
Hmm, the mocks seem to have the buttons separated by 8px (https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=7941%3A18814) 

... and also both Plain buttons. Should I revert them back to Plain buttons or keep the "actionable" button as Primary?

Reply:
Hmm odd. Let's leave as-is then but create a follow-up issue to make things more consistent.

Opened #2080 to make comparison easier.

Reply:
Opened #2104 as follow up issue :)